modules/dnn/test/test_layers_1d.cpp

```diff
@@ -245,4 +245,52 @@ INSTANTIATE_TEST_CASE_P(/*nothing*/, Layer_Elemwise_1d_Test, Combine(
 /*operation*/           Values("div", "prod", "max", "min", "sum")
 ));
 
+// WORKING FOR 1D
+typedef testing::TestWithParam<tuple<int>> Layer_Split_Test;
+TEST_P(Layer_Split_Test, Accuracy)
+{
+    LayerParams lp;
+    lp.type = "Split";
+    lp.name = "SplitLayer";
+    int top_count = 2; // 2 is for simplicity
+    lp.set("top_count", top_count);
+
+    int dims = get<0>(GetParam());
+
+    Ptr<SplitLayer> layer = SplitLayer::create(lp);
+    std::vector<int> input_shape;
+    std::vector<int> output_shape;
+    Mat input;
+
+    if (dims > 1){
+        input_shape = {1, 10}; // 10 to keep test fast
+        output_shape = {1, 10};
+        input = Mat(input_shape, CV_32F);
+    } else {
+        input_shape = {dims};
+        output_shape = {dims};
+        input = Mat(dims, input_shape.data(), CV_32F);
+    }
+
+    cv::randn(input, 0.0, 1.0);
+
+    Mat output_ref;
+    if (dims > 1){
+        output_ref = Mat(output_shape, CV_32F, input.data);
+    }else{
+        output_ref = Mat(dims, output_shape.data(), CV_32F, input.data);
+    }
```

Comment:
@vpisarev What is wrong with Mat API if we need to do that?

Reply:
No thing is wrong. I have changed initialization scheme.